components/proxy/src/main/java/com/hotels/styx/proxy/BackendServicesRouter.java

```diff
@@ -178,10 +180,11 @@ private OriginHealthStatusMonitor healthStatusMonitor(BackendService backendServ
                                 healthCheckClient);
     }
 
-    private StyxHttpClient healthCheckClient(BackendService backendService) {
+    private StyxHttpClient healthCheckClient(BackendService backendService, HttpMessageFormatter httpMessageFormatter) {
         StyxHttpClient.Builder builder = new StyxHttpClient.Builder()
                 .connectTimeout(backendService.connectionPoolConfig().connectTimeoutMillis(), MILLISECONDS)
-                .userAgent("Styx/" + environment.buildInfo().releaseVersion());
+                .userAgent("Styx/" + environment.buildInfo().releaseVersion())
+                .httpMessageFormatter(httpMessageFormatter);
```

Comment:
Health check client (functionality) doesn't really need message formatting. Of course there could be a technical reason why this is passed in here. I didn't check.

Reply:
This is needed because the health check client constructs an instance of HttpRequestOperation which in turn constructs an instance of HttpRequestMessageLogger.  This is old code, but I think we should consider injecting HttpRequestMessageLogger instead of constructing it in two places.  Ultimately all 'new' operations should be performed early and we should try(where possible) to construct our objects all in one place.

Reply:
reverted as no longer needed as we have a default now